### PR TITLE
[MB-2531] Add response info to notification event

### PR DESCRIPTION
**Pull request by rlepinski**
Adds notification response info to the open event.

I also fixed a few issues when I was testing. Lots of main thread checker warnings for iOS and build issues for Android. 

Testing:
 - Android registration event, notification opens, etc..
- iOS registration event. Still need to test a interactive notification.

### CODE REVIEW

File: src/android/events/NotificationOpenedEvent.java [hunk 1]
```diff
@@ -41,17 +45,53 @@
...
+
+
+    @Override
+    public JSONObject getEventData() {
+        JSONObject jsonObject = super.getEventData();
+
+        try {
+            if (actionButtonInfo != null) {
+                jsonObject.put(ACTION_ID, actionButtonInfo.getButtonId());
+                jsonObject.put(IS_FOREGROUND, actionButtonInfo.isForeground());
```
Comment (marc-scig) [inline src/android/events/NotificationOpenedEvent.java]: do we need `IS_FOREGROUND` in the event data here? slash where's this stuff spec'ed out?

File: src/ios/UAirshipPlugin.m [hunk 2]
```diff
@@ -237,24 +237,22 @@ - (CDVPluginResult *)pluginResultForValue:(id)value status:(CDVCommandStatus)sta
...
  * @param command The cordova command.
  * @param block The UACordovaExecutionBlock to execute.
  */
 - (void)performCallbackWithCommand:(CDVInvokedUrlCommand *)command withBlock:(UACordovaExecutionBlock)block {
-    [self.commandDelegate runInBackground:^{
-        UACordovaCompletionHandler completionHandler = ^(CDVCommandStatus status, id value) {
-            CDVPluginResult *result = [self pluginResultForValue:value status:status];
-            [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
-        };
+    UACordovaCompletionHandler completionHandler = ^(CDVCommandStatus status, id value) {
```
Comment (marc-scig) [inline src/ios/UAirshipPlugin.m]: is this related to the "main thread checker" bit you were mentioning? does this effectively make all cordova callbacks run on the UI thread?

File: src/ios/UAirshipPlugin.m [hunk 3]
```diff
@@ -1014,4 +1009,40 @@ -(NSInteger)parseLogLevel:(NSString *)logLevel defaultLogLevel:(NSInteger)defaul
     return defaultValue;
 }
 
+
+- (UANotificationAction *)notificationActionForCategory:(NSString *)category actionIdentifier:(NSString *)identifier {
```
Comment (marc-scig) [inline src/ios/UAirshipPlugin.m]: this looks like a duplication of the class method in UAAppIntegration. Any chance we could use that instead?

File: src/ios/UAirshipPlugin.m [hunk 4]
```diff
@@ -791,18 +799,14 @@ - (void)showInbox {
 
 - (void)displayMessageCenter:(CDVInvokedUrlCommand *)command {
     [self performCallbackWithCommand:command withBlock:^(NSArray *args, UACordovaCompletionHandler completionHandler) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [[UAirship defaultMessageCenter] display];
-        });
+        [[UAirship defaultMessageCenter] display];
```
Comment (BrianBatchelder) [inline src/ios/UAirshipPlugin.m]: Why did this have to change? I thought the warning was about doing stuff _off_ the main thread.

File: src/ios/UAirshipPlugin.m (hunk 2 above)
Comment (rlepinski) [reply src/ios/UAirshipPlugin.m]: Yes, but all of ours calls are none blocking

File: src/ios/UAirshipPlugin.m (hunk 4 above)
Comment (rlepinski) [reply src/ios/UAirshipPlugin.m]: We no longer call everything on a background thread

File: src/ios/UAirshipPlugin.m (hunk 3 above)
Comment (rlepinski) [reply src/ios/UAirshipPlugin.m]: We could if it was public. Right now its internal only.

File: src/android/events/NotificationOpenedEvent.java (hunk 1 above)
Comment (rlepinski) [reply src/android/events/NotificationOpenedEvent.java]: IS_FOREGROUND is if the notification response is a foreground or background action.

File: src/ios/UAirshipPlugin.m (hunk 4 above)
Comment (BrianBatchelder) [reply src/ios/UAirshipPlugin.m]: OK.

**Comment by marc-scig** [comment]
🐻

**Pull request merged**